stdlib/public/Concurrency/Task.cpp

```diff
@@ -83,13 +182,35 @@ AsyncTaskAndContext
 swift::swift_task_create_f(JobFlags flags, AsyncTask *parent,
                            AsyncFunctionType<void()> *function,
                            size_t initialContextSize) {
-  assert(!flags.task_isFuture() && "function doesn't support creating futures");
+  return swift_task_create_future_f(
+      flags, parent, nullptr, function, initialContextSize);
+}
+
+AsyncTaskAndContext swift::swift_task_create_future(
+    JobFlags flags, AsyncTask *parent, const Metadata *futureResultType,
+    const AsyncFunctionPointer<void()> *function) {
+  return swift_task_create_future_f(
+      flags, parent, futureResultType, function->Function.get(),
+      function->ExpectedContextSize);
+}
+
+AsyncTaskAndContext swift::swift_task_create_future_f(
+    JobFlags flags, AsyncTask *parent, const Metadata *futureResultType,
+    AsyncFunctionType<void()> *function, size_t initialContextSize) {
+  assert((futureResultType != nullptr) == flags.task_isFuture());
+  assert((futureResultType != nullptr) == flags.task_isFuture());
```

Comment:
Seeing undouble?

Should we assert here that `initialContextSize >= sizeof(FutureAsyncContext)`?

Reply:
That's a good idea!